crates/driver/src/infra/mempool/mod.rs

```diff
@@ -0,0 +1,68 @@
+use {
+    crate::{
+        domain::eth,
+        infra::blockchain::{self, Ethereum},
+    },
+    futures::future::join_all,
+};
+
+/// The mempool to use for publishing onchain transactions. The public mempool
+/// of an [`Ethereum`] node can be used, or one of the private mempools offered
+/// by various transaction relay services.
+#[derive(Debug, Clone)]
+pub struct Mempool(Inner);
```

Comment:
I'm not a fan of moving this into the domain just yet. Executing transactions on chain seems easy (just use the `eth_sendTransaction` RPC call right?), but it is actually super difficult to get right and made reliable.

I would definitely wrap our current settlement submission logic in a `boundary` module for transaction submitting. To name a couple of the things that we need to support:
- Re-broadcasting the transaction with higher gas prices if it does not get included
- Sending cancellation transactions if we go over the deadline of the solution stops submitting successfully in order to save on gas costs